src/components/GlobalSearch/GlobalSearchComponent.tsx

```diff
@@ -0,0 +1,262 @@
+      <ResourceSectionHeader>{header.toUpperCase()}</ResourceSectionHeader>
+    ) : null;
+  const itemsGroupedByResourceType = _.groupBy(
+    items,
+    item => item.resourceType
+  );
+  const renderItemsIfExists = (items: SearchResource[]) => {
+    const itemsOrEmpty = items != null ? items : [];
+    return (
+      <SearchResultList
+        strings={strings}
+        onItemSelected={onItemSelected}
+        items={itemsOrEmpty}
+        renderSearchItem={renderSearchItem}
+      ></SearchResultList>
+    );
+  };
+
+  return (
+    <>
+      {renderHeaderIfTrue(searchForFilter.assets, strings.assets)}
+      {renderItemsIfExists(itemsGroupedByResourceType.asset)}
+      {renderHeaderIfTrue(searchForFilter.timeSeries, strings.timeseries)}
+      {renderItemsIfExists(itemsGroupedByResourceType.timeSeries)}
+      {renderHeaderIfTrue(searchForFilter.events, strings.events)}
+      {renderItemsIfExists(itemsGroupedByResourceType.event)}
+      {renderHeaderIfTrue(searchForFilter.files, strings.files)}
+      {renderItemsIfExists(itemsGroupedByResourceType.file)}
+    </>
+  );
+}
+
+export function GlobalSearch({
+  strings: userStrings,
+  searchForFilter: userSearchForFilter,
+  onError,
+  renderSearchResult: userRenderSearchResult,
+  renderSearchItem: userRenderSearchItem,
+  onItemSelected: userOnItemSelected,
+  debounceTime,
+  onSearchResults,
+}: GlobalSearchProps) {
+  const [searchResults, setSearchResults] = useState<SearchResource[] | null>(
+    null
+  );
+  const [loading, setLoading] = useState<boolean>(false);
+  const [searchCounter] = useState<RefCounter>(new RefCounter());
+  const client = useContext(ClientSDKContext);
+  const strings = { ...defaultStrings, ...(userStrings || {}) };
+  const searchForFilter = {
+    ...{
+      assets: true,
+      events: true,
+      timeSeries: true,
+      files: true,
+    },
+    ...(userSearchForFilter || {}),
+  };
+
+  const renderSearchResult =
+    userRenderSearchResult || defaultRenderSearchResult;
+  const renderSearchItem = userRenderSearchItem || defaultRenderSearchItem;
+  const onItemSelected: ItemSelectedCallback = item => {
+    if (userOnItemSelected) {
+      userOnItemSelected(item);
+    }
+  };
+
+  useEffect(() => {
+    if (onSearchResults && searchResults != null) {
+      onSearchResults(searchResults);
+    }
+  }, [searchResults]);
+
+  const handleSearch = useCallback(
+    (searchQuery: ApiQuery) => {
+      if (client == null) {
+        throw new Error(
+          'Remember to add a ClientSDKProvider before using any Gearbox components.'
+        );
+      }
+
+      if (searchQuery.query == null || searchQuery.query === '') {
+        setSearchResults(null);
+        return;
+      }
+
+      setLoading(true);
+      // The searchCounter mitigates old search results from replacing newer search results
+      const thisId = searchCounter.nextId();
+      globalSearch(client, searchQuery.query, searchForFilter, 50)
+        .then(x => {
```

Comment:
can use await/catch instead?

Reply:
True it might be allowed in a `useCallback` hook, but for `useEffect` you have to create an inner function to do so because it expects a deconstructor function returned.

Reply:
I see

Reply:
In `useCallback` it should be fine, since the callback function isn't expected to return anything, so I'm going for async/await